oneview_redfish_toolkit/blueprints/subscription.py

```diff
@@ -0,0 +1,104 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2017) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import logging
+
+from flask import abort
+from flask import Blueprint
+from flask import request
+from flask import Response
+from flask_api import status
+
+from oneview_redfish_toolkit.api.errors import OneViewRedfishError
+from oneview_redfish_toolkit.api.subscription import Subscription
+from oneview_redfish_toolkit import util
+
+
+subscription = Blueprint("subscription", __name__)
+
+
+@subscription.route(
+    "/redfish/v1/EventService/EventSubscriptions/", methods=["POST"])
+def add_subscription():
+    """Add the Redfish Subscription.
+
+        Add a new subscription when this POST operation is requested.
+        The body of the request must have an array of EventTypes.
+
+        EventTypes:
+            - StatusChange
+            - ResourceUpdated
+            - ResourceAdded
+            - ResourceRemoved
+            - Alert
+
+        Returns:
+            JSON: JSON with Subscription information.
+
+        Exception:
+            OneViewRedfishError: When occur a key mapping error.
+            return abort(400)
+
+            Exception: Unexpected error.
+            return abort(500)
+    """
+    try:
+        # get information from post
+        # add subscription in subscriptions_by_type
+        # add subscription in all_subscriptions
+        # the subscription id is len(all_subscriptions) + 1
+
+        try:
+            body = request.get_json()
+            event_types = body["EventTypes"]
+        except Exception:
+            raise OneViewRedfishError(
+                {"errorCode": "INVALID_INFORMATION",
+                 "message": "Invalid JSON key. The JSON request body"
+                            " must have the key EventTypes."})
+
+        subscription_id = str(len(util.all_subscriptions) + 1)
```

Comment:
Why? Can't we use UUID as all the other resources?

Reply:
The required value is an `Id`, that according the spec it is different than `UUID`. Should we generate a random `UUID`?

Reply:
I know... from the spec the `Id` is a string, so we are allowed to store whatever we want in it, as we've been doing for all resources implemented until this point. If not, then we should think of a new strategy to fill this for ComputerSystem, Chassis, Manager, etc...

If we agree on using UUID, then in my opinion generate a new one for this field is our best and safest approach. Using incremental IDs as it is implemented here will lead to error right after deleting a subscription...

Reply:
Ok. So, I'm using the Python UUID lib to generate it. As I'm using the `uuid1` it has zero chances of collision with a previous generated value, since it takes data/time.